sdk/core/azure-core/src/main/java/com/azure/core/credential/BasicAuthenticationCredential.java

```diff
@@ -32,24 +25,15 @@
      * @param password basic auth password
      */
     public BasicAuthenticationCredential(String username, String password) {
-        this.username = username;
-        this.password = password;
+        String credential = username + ":" + password;
+        this.encodedCredential = Base64Util.encodeToString(credential.getBytes(StandardCharsets.UTF_8));
     }
 
     /**
      * @throws RuntimeException If the UTF-8 encoding isn't supported.
      */
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        String credential = username + ":" + password;
-        String encodedCredential;
-        try {
-            encodedCredential = Base64Util.encodeToString(credential.getBytes("UTF8"));
-        } catch (UnsupportedEncodingException e) {
-            // The encoding is hard-coded, so if it's unsupported, it needs to be fixed right here.
-            throw logger.logExceptionAsError(new RuntimeException(e));
-        }
-
-        return Mono.just(new AccessToken(encodedCredential, OffsetDateTime.MAX));
+        return Mono.fromCallable(() -> new AccessToken(encodedCredential, OffsetDateTime.MAX));
```

Comment:
would `Mono.just` work?
```suggestion
        return Mono.just(new AccessToken(encodedCredential, OffsetDateTime.MAX));
```

Reply:
Yes, `Mono.just` would work but `Mono.fromCallable` further defers the creation of `AccessToken`, so if the reactive stream is never subscribed the `AccessToken` is never instantiated.